src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala

```diff
@@ -450,21 +451,26 @@ trait SilverTransforms extends SparkSessionWrapper {
 
     val clusterBaseDF = clusterBase(df)
 
-    // TODO -- Issue_37 -- instance pool lookup pass in from bronze
+    val poolSnapLookup =  pools_snapshot.asDF
+      .withColumn("timestamp", unix_timestamp('Pipeline_SnapTS) * 1000)
+      .select('timestamp, 'organization_id, 'instance_pool_id, 'node_type_id.alias("pool_snap_node_type"))
+
     val instancePoolLookup = auditRawTable.asDF
       .filter('serviceName === "instancePools" && 'actionName === "create")
       .select(
-        'organization_id, get_json_object($"response.result", "$.instance_pool_id").alias("instance_pool_id"),
+        'timestamp, 'organization_id, get_json_object($"response.result", "$.instance_pool_id").alias("instance_pool_id"),
         $"requestParams.node_type_id".alias("pool_node_type")
       )
 
-    val filledDriverType = when('cluster_name.like("job-%-run-%"), coalesce('driver_node_type_id, 'node_type_id))
-      .when('instance_pool_id.isNotNull, 'pool_node_type)
-      .when(isSingleNode, 'node_type_id)
-      .otherwise(coalesce('driver_node_type_id, first('driver_node_type_id, true).over(clusterBefore), 'node_type_id))
+    // Issue_37 -- this will need to be reviewed for mixed pools (i.e. driver of different type)
+    val filledDriverType =
+      when('instance_pool_id.isNotNull, coalesce('pool_node_type, 'pool_snap_node_type)) // both driver and worker nodes are same when using pool
+        .when('cluster_name.like("job-%-run-%"), coalesce('driver_node_type_id, 'node_type_id)) // when jobs clusters workers == driver driver node type is not defined
```

Comment:
I think that it will not work correctly with the hybrid pools: we need to check first the `driver_instance_pool_id`

Reply:
good call, fixed